### PR TITLE
Improve gitpod config

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

**Comment by timja** [comment]
Routing is only partly working and maven caching is not happening

**Comment by timja** [comment]
I enabled this feature let's see:
image

**Comment by timja** [comment]
FYI @Nancy-Chauhan

**Comment by timja** [comment]
Still redirecting wrong so I assume `getRootUrlFromRequest` is failing

Here's a dump of headers from server side:

```
X-Forwarded-Host: 8080-jenkinsci-designlibrary-1qcpytmyc1p.ws-eu45.gitpod.io:443
X-Forwarded-Proto: https
```

which looks fine

**Comment by timja** [comment]
Debug in admin monitor logs:

```
coming from https://8080-jenkinsci-designlibrary-1qcpytmyc1p.ws-eu45.gitpod.io/jenkins/manage and redirecting to https://8080-jenkinsci-designlibrary-1qcpytmyc1p.ws-eu45.gitpod.io/jenkins/administrativeMonitor/hudson.diagnosis.ReverseProxySetupMonitor/testForReverseProxySetup/https%3A%2F%2F8080-jenkinsci-designlibrary-1qcpytmyc1p.ws-eu45.gitpod.io%2Fjenkins%2Fmanage/
```

https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/diagnosis/ReverseProxySetupMonitor.java#L88

**Comment by Nancy-Chauhan** [comment]
> Routing is only partly working and maven caching is not happening

Hey @timja , regarding "maven caching". This will happen if prebuilds are enabled in Jenkins otherwise data will be cached only for a workspace restart.

**Comment by timja** [comment]
> > Routing is only partly working and maven caching is not happening
> 
> Hey @timja , regarding "maven caching". This will happen if prebuilds are enabled in Jenkins otherwise data will be cached only for a workspace restart.

It seems to somewhat work. but yeah 95% of the time is downloading deps.

It takes 6 seconds with cache or minutes without.

Not that big of a deal though.

----

The main issue is that redirects are broken because Jenkins thinks it's running under http://localhost:8080/jenkins/

**Comment by timja** [comment]
I'll ship this as it improves the current state

File: .gitpod.yml [hunk 1]
```diff
@@ -1,8 +1,16 @@
 tasks:
-  - init: mvn install -P quick-build
+  - init: |
+      mvn install -P quick-build
+      curl https://gist.githubusercontent.com/timja/fe73f0a0f98afe4d9356e7ce6752d95b/raw/a8d772267bed81fb0c11af2c69eaa71ebc7daec0/init.sh | bash
```
Comment (akosyakov) [inline .gitpod.yml:4]: you are better to move it out of [init](https://www.gitpod.io/docs/references/gitpod-yml#tasksninit).

> When you start a workspace that does not have a [prebuild](https://www.gitpod.io/docs/prebuilds), init is executed at workspace start. When you start a workspace that has a prebuild, init executes as part of the prebuild, but **does NOT execute again** at workspace start.

Comment (akosyakov) [reply .gitpod.yml:4]:
```yaml
tasks:
  - init: mvn install -P quick-build
    command: |
      curl https://gist.githubusercontent.com/timja/fe73f0a0f98afe4d9356e7ce6752d95b/raw/a8d772267bed81fb0c11af2c69eaa71ebc7daec0/init.sh | bash
      mvn hpi:run
    name: Run
  - command: gp await-port 8080 && gp url 8080 && gp preview $(gp url 8080)/jenkins/
    name: Preview
```

Comment (timja) [reply .gitpod.yml:4]: https://github.com/jenkinsci/design-library-plugin/pull/66

**Pull request merged**